File: JS/2185.counting_words_with_a_given_prefix [2].js
/**
 * @param {string[]} words
 * @param {string} pref
 * @return {number}
 */
var prefixCount = function(words, pref) {
    return words.reduce((acc, word) => acc + (word.startsWith(pref) ? 1 : 0), 0)
};

File: CPP [2001-2500]/2185.counting-words-with-a-given-prefix [2].cpp
class Solution {
public:
    // T.C.=O(n*m), S.C.=O(m)
    // n: size of words[], m: length of pref
    int prefixCount(vector<string>& words, string pref) {
        int count = 0;
        for(string& word: words)
            if(word.substr(0, pref.length()) == pref)
                count++;

        return count;
    }
};